mne/beamformer/tests/test_lcmv.py

```diff
@@ -688,6 +689,61 @@ def test_lcmv_reg_proj(proj):
                         weight_norm='nai', rank=None, verbose=True)
     want_rank = 302  # 305 good channels - 3 MEG projs
     assert filters['rank'] == want_rank
+    # And also with and without noise_cov
+    with pytest.raises(ValueError, match='several sensor types'):
+        make_lcmv(epochs.info, forward, data_cov, reg=0.05,
+                  noise_cov=None)
+    epochs.pick_types('grad')
+    kwargs = dict(reg=0.05, pick_ori=None, weight_norm=weight_norm)
+    filters_cov = make_lcmv(epochs.info, forward, data_cov,
+                            noise_cov=noise_cov, **kwargs)
+    filters_nocov = make_lcmv(epochs.info, forward, data_cov,
+                              noise_cov=None, **kwargs)
+    ad_hoc = mne.make_ad_hoc_cov(epochs.info)
+    filters_adhoc = make_lcmv(epochs.info, forward, data_cov,
+                              noise_cov=ad_hoc, **kwargs)
+    evoked = epochs.average()
+    stc_cov = apply_lcmv(evoked, filters_cov)
+    stc_nocov = apply_lcmv(evoked, filters_nocov)
+    stc_adhoc = apply_lcmv(evoked, filters_adhoc)
+
+    # Compare adhoc and nocov: scale difference is necessitated by using std=1.
+    if weight_norm == 'unit-noise-gain':
+        scale = np.sqrt(ad_hoc['data'][0])
+    else:
+        scale = 1.
+    assert_allclose(stc_nocov.data, stc_adhoc.data * scale)
+    assert_allclose(
+        np.dot(filters_nocov['weights'], filters_nocov['whitener']),
+        np.dot(filters_adhoc['weights'], filters_adhoc['whitener']) * scale)
+
+    # Compare adhoc and cov: locs might not be equivalent, but the same
+    # general profile should persist, so look at the std and be lenient:
+    if weight_norm == 'unit-noise-gain':
+        adhoc_scale = 0.12
+    else:
+        adhoc_scale = 1.
+    assert_allclose(
+        np.linalg.norm(stc_adhoc.data, axis=0) * adhoc_scale,
+        np.linalg.norm(stc_cov.data, axis=0), rtol=0.3)
+    assert_allclose(
+        np.linalg.norm(stc_nocov.data, axis=0) / scale * adhoc_scale,
+        np.linalg.norm(stc_cov.data, axis=0), rtol=0.3)
+
+    if weight_norm == 'nai':
+        # NAI always represents something normalized
+        for stc in (stc_nocov, stc_cov):
+            assert_allclose(stc.data.std(), 0.39, rtol=0.1)
+    elif weight_norm is None:
+        # None always represents something not normalized, reflecting channel
+        # weights
+        for stc in (stc_nocov, stc_cov):
+            assert_allclose(stc.data.std(), 1.4e-8, rtol=0.1)
+    else:
+        assert weight_norm == 'unit-noise-gain'
+        # unit-noise-gain normalization depends on `noise_cov`
```

Comment:
As outlined earlier in the comments, this is not right. The whitening changes the output values, but unit-noise-gain does not depend on ``noise-cov``.